pennylane/templates/tensornetworks/mps.py

```diff
@@ -0,0 +1,82 @@
+import pennylane as qml
+import pennylane.numpy as np
+from pennylane.operation import Operation, AnyWires
+import re
+
+
+def compute_indices_MPS(wires, loc):
+    """
+    Generate a list of wire indices that quantum gates acts on
+    Args:
+        loc (int): local wire number of a single quantum gate
+        wires (Iterable): the wires on which MPS acts
+    Returns:
+        layers (array): array of wire indices or wire labels for each block
+    """
+    
+    layers = np.array([[wires[idx] for idx in range(j, j+loc)] for j in range(0, len(wires) - loc // 2, loc // 2)])
+    return layers
+
+
+class MPS_from_function(Operation):
+
+    num_params = 1
+    num_wires = AnyWires
+    par_domain = "A"
+
+    def __init__(
+        self,
+        wires,
+        loc,
+        block,
+        n_params_block,
+        weights=None,
+        do_queue=True,
+        id=None,
+    ):
+        n_wires = len(wires)
+        assert loc >= 2, f"loc must be larger than or equal to 2; got {loc}"
+        assert n_wires >= 3, f"number of wires must be greater than or equal to 3; got {n_wires}"
+        assert loc <= n_wires, f"loc must be smaller than or equal to the number of wires; got loc = {loc} and number of wires = {n_wires}"
+  
+        self.n_blocks = int(n_wires / (loc / 2) - 1)
+        self.block = block
+
```

Comment:
Codecov is another continuous integration tool whose role is to automatically check that all code is properly tested. Please address all these issues raised on missing tests